include/deal.II/boost_adaptors/bounding_box.h

```diff
@@ -71,7 +71,11 @@ namespace boost
         static inline void
         set(dealii::BoundingBox<dim, Number> &box, Number value)
         {
-          box.get_boundary_points().first[D] = value;
+          std::pair<dealii::Point<dim, Number>, dealii::Point<dim, Number>>
+            corner_points        = box.get_boundary_points();
+          corner_points.first[D] = value;
+
+          box = dealii::BoundingBox<dim, Number>(corner_points);
```

Comment:
Since we cannot guarantee that the value will actually be smaller than `corner_points.second[D]`, I think we should do something like
```
if (value > corner_points.second[D])
   corner_points.second[D] = value;
else 
   corner_points.first[D] = value;
```